src/gurt/misc.c

```diff
@@ -947,149 +951,451 @@ d_rank_range_list_free(d_rank_range_list_t *range_list)
 }
 
 static inline bool
-dis_integer_str(char *str)
+dis_unsigned_str(char *str)
 {
-	char	*p;
+	char *eos;
 
-	p = str;
-	if (p == NULL || strlen(p) == 0)
+	if (str == NULL || str[0] == '\0')
 		return false;
 
-	while (*p != '\0') {
-		if (*p <= '9' && *p >= '0') {
-			p++;
-			continue;
-		} else {
-			return false;
-		}
-	}
+	eos = str + (sizeof(UINT64_MAX_STR) - 1);
+	while (str != eos && *str != '\0' && *str >= '0' && *str <= '9')
+		++str;
 
-	return true;
+	return *str == '\0';
 }
 
 static inline bool
 dis_single_char_str(char *str)
 {
-	return strlen(str) == 1;
+	return strnlen(str, 2) == 1;
+}
+
+/**
+ * Overloads to hook the unsafe getenv()/[un]setenv()/putenv()/clearenv()
+ * functions from glibc.
+ */
+static pthread_rwlock_t d_env_lock = PTHREAD_RWLOCK_INITIALIZER;
+
+/**
+ * Check if and environment variable is defined.
+ *
+ * \param[in]		name		name of the environment variable.
+ * \return				true iff the environment variable is defined.
+ */
+bool
+d_isenv_def(char *name)
+{
+	char *env;
+
+	pthread_rwlock_rdlock(&d_env_lock);
```

Comment:
These will be coverity warnings unless you either check the return code or use our macros.

Reply:
I did wonder.  Does the logging ever call getenv though?

I made the same change to the locking in the logging as that was causing recursion and found out about the coverity issue that way.

Reply:
From what I remember, I experiment infinite recursion with the function `d_logfac_is_enabled()` from `src/gurt/dlog.c`.  There is other functions from this file which could also probably lead to the same kind of issues.

Reply:
> Thanks for the info. Sadly, I will not be able to use the D_RWLOCK_RDLOCK macros as it use the DAOS logging facilities which could lead to infinite recursive calls. Thus I will check the return code is a similar way as this last macros.
> 
> * [x]  Checks return code when using standard function calls such as `pthread_rwlock_rdlock()`

Fixed with commit 2791b9323afa26af021220b35b0cffee1e2a0a63